katran/lib/IOBufWriter.h

```diff
@@ -0,0 +1,47 @@
+/*
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+#pragma once
+
+#include <memory>
+#include <string>
+#include <folly/io/IOBuf.h>
+
+#include "katran/lib/DataWriter.h"
+
+namespace katran {
+
+/**
+ * IOBufWriter is used to write pcap-data into IOBuf.
+ */
+class IOBufWriter : public DataWriter {
+ public:
+  /**
+   * @param unique_ptr<IOBuf> iobuf for packets to written into
+   */
+  explicit IOBufWriter(folly::IOBuf* iobuf);
```

Comment:
Why not give ownership of IOBuf to IOBufWriter?

Reply:
because KatranMonitor needs to have a reference to em as well (so getEventBuffer helper could be called.  somehow we need to wire all of this IOBuf all the way back to KatranLb)